File: elixir/sum-of-multiples/sum_of_multiples.exs
defmodule SumOfMultiples do
  @doc """
  Adds up all numbers from 1 to a given end number that are multiples of the factors provided.
  """
  @spec to(non_neg_integer, [non_neg_integer]) :: non_neg_integer
  def to(limit, list) do
    Enum.filter((limit - 1)..2, fn(n) ->
      Enum.any?(list, fn(l) -> rem(n, l) == 0 end)
    end)
    |> Enum.sum
  end
end

File: ruby/sum-of-multiples/sum_of_multiples.rb
class SumOfMultiples

  attr_accessor :divisors

  def initialize(*numbers)
    @divisors = numbers
  end

  def to(limit)
    (1...limit).select do |number|
      divisors.any? do |divisor|
        (number % divisor).zero?
      end
    end.sum
  end
end
